src/inabox/inabox-viewport.js

```diff
@@ -81,11 +85,21 @@ export class ViewportBindingInabox {
     // 2) broadcast the request
     this.iframeClient_.setHostWindow(win.top);
 
+    /** @private {boolean} */
+    this.visibilityV2Enabled_ =
+        nativeIntersectionObserverSupported(win) &&
+            isExperimentOn(win, 'visibility-v2');
+
     dev().fine(TAG, 'initialized inabox viewport');
   }
 
   /** @override */
   connect() {
+    if (this.visibilityV2Enabled_) {
+      // Visibility V2 uses native IntersectionObserver, no position data needed
```

Comment:
But we will update `this.viewportRect` in the polyfill case, and then runtime will take care of the scheduling layout of the element. I can't find where we update this in our callback function. Without it how would runtime know when to layout the element of inabox? This also happen to the viewport `onScroll` and viewport `onResize` event.  Am I missing something?